src/win/util.c

```diff
@@ -1283,3 +1283,101 @@ int uv_os_tmpdir(char* buffer, size_t* size) {
 }
+
+
+void uv_free_passwd(uv_passwd_t* pwd) {
+  if (pwd == NULL)
+    return;
+
+  uv__free(pwd->name);
+  uv__free(pwd->shell);
+  uv__free(pwd->homedir);
+}
+
+
+int uv_get_passwd(uv_passwd_t* pwd) {
+  HANDLE token;
+  wchar_t username[UNLEN + 1];
+  wchar_t path[MAX_PATH];
+  DWORD bufsize;
+  int r;
+
+#define CONVERT_BUFFER(src, dst)                                              \
+  do {                                                                        \
+    /* Check how much space we need */                                        \
+    bufsize = WideCharToMultiByte(CP_UTF8, 0, src, -1, NULL, 0, NULL, NULL);  \
+                                                                              \
+    if (bufsize == 0) {                                                       \
+      uv_free_passwd(pwd);                                                    \
+      return uv_translate_sys_error(GetLastError());                          \
+    }                                                                         \
+                                                                              \
+    /* Allocate the destination buffer */                                     \
+    dst = uv__malloc(bufsize);                                                \
+                                                                              \
+    if (dst == NULL) {                                                        \
+      uv_free_passwd(pwd);                                                    \
+      return UV_ENOMEM;                                                       \
+    }                                                                         \
+                                                                              \
+    /* Convert to UTF-8 */                                                    \
+    bufsize = WideCharToMultiByte(CP_UTF8,                                    \
+                                  0,                                          \
+                                  src,                                        \
+                                  -1,                                         \
+                                  dst,                                        \
+                                  bufsize,                                    \
+                                  NULL,                                       \
+                                  NULL);                                      \
+                                                                              \
+    if (bufsize == 0) {                                                       \
+      uv_free_passwd(pwd);                                                    \
+      return uv_translate_sys_error(GetLastError());                          \
+    }                                                                         \
+  }                                                                           \
+  while (0)
```

Comment:
Does this need to be a macro?  Also, if a macro contains control flow, it's good form to advertise that through the macro name (e.g. `TRY_CONVERT_BUFFER_OR_RETURN(...)`.)